code/modules/projectiles/updated_projectiles/guns/energy.dm

```diff
@@ -242,3 +276,276 @@
+	if(flags_gun_features & GUN_BURST_FIRING)
+		return
+
+	if(istype(I,/obj/item/attachable))
+		if(check_inactive_hand(user))
+			attach_to_gun(user,I)
+
+ 	//the active attachment is reloadable
+	else if(active_attachable && active_attachable.flags_attach_features & ATTACH_RELOADABLE)
+		if(!check_inactive_hand(user))
+			return
+		if(istype(I,/obj/item/ammo_magazine))
+			var/obj/item/ammo_magazine/MG = I
+			if(istype(src, MG.gun_type))
+				to_chat(user, "<span class='notice'>You disable [active_attachable].</span>")
+				playsound(user, active_attachable.activation_sound, 15, 1)
+				active_attachable.activate_attachment(src, null, TRUE)
+				reload(user,MG)
+				return
+		active_attachable.reload_attachment(I, user)
+
+	else if(istype(I,/obj/item/cell/lasgun))
+		if(check_inactive_hand(user))
+			reload(user,I)
+
+
+/obj/item/weapon/gun/energy/lasgun/reload(mob/user, obj/item/cell/lasgun/new_cell)
+	if(flags_gun_features & (GUN_BURST_FIRING|GUN_UNUSUAL_DESIGN|GUN_INTERNAL_MAG))
+		return
+
+	if(!new_cell || !istype(new_cell))
+		to_chat(user, "<span class='warning'>That's not a power cell!</span>")
+		return
+
+	if(new_cell.charge <= 0)
+		to_chat(user, "<span class='warning'>[cell] is depleted!</span>")
+		return
+
+	if(!istype(src, new_cell.gun_type))
+		to_chat(user, "<span class='warning'>That power cell doesn't fit in there!</span>")
+		return
+
+	if(cell)
+		to_chat(user, "<span class='warning'>It's still got something loaded.</span>")
+		return
+
+	if(user)
+		if(new_cell.reload_delay > 1)
+			to_chat(user, "<span class='notice'>You begin reloading [src]. Hold still...</span>")
+			if(do_after(user,new_cell.reload_delay, TRUE, 5, BUSY_ICON_FRIENDLY))
+				replace_magazine(user, new_cell)
+			else
+				to_chat(user, "<span class='warning'>Your reload was interrupted!</span>")
+				return
+		else
+			replace_magazine(user, new_cell)
+	else
+		replace_magazine(null, new_cell)
+	return TRUE
+
+/obj/item/weapon/gun/energy/lasgun/replace_magazine(mob/user, obj/item/cell/lasgun/new_cell)
+	cell = new_cell
+	if(user)
+		user.drop_inv_item_to_loc(new_cell, src) //Click!
+		user.visible_message("<span class='notice'>[user] loads [new_cell] into [src]!</span>",
+		"<span class='notice'>You load [new_cell] into [src]!</span>", null, 3)
+		if(reload_sound)
+			playsound(user, reload_sound, 25, 1, 5)
+		update_icon(user)
+	else
+		cell.loc = src
+		update_icon()
+
+//Drop out the magazine. Keep the ammo type for next time so we don't need to replace it every time.
+//This can be passed with a null user, so we need to check for that as well.
+/obj/item/weapon/gun/energy/lasgun/unload(mob/user, reload_override = 0, drop_override = 0) //Override for reloading mags after shooting, so it doesn't interrupt burst. Drop is for dropping the magazine on the ground.
+	if(!reload_override && (flags_gun_features & (GUN_BURST_FIRING|GUN_UNUSUAL_DESIGN|GUN_INTERNAL_MAG)))
+		return
+
+	if(!cell || isnull(cell) || cell.loc != src)
```

Comment:
Isn't `!cell || isnull(cell)` essentially the same?

Reply:
I believe similar wording was used in the source code that was adapted so I kept it